### PR TITLE
ie: guardrails without dml

### PR DESCRIPTION
**Pull request by tomhoule**
Only the models in this commit.

### CODE REVIEW

File: introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs [hunk 1]
```diff
@@ -8,6 +8,9 @@ pub(super) fn introspect_models(datamodel: &mut dml::Datamodel, ctx: &mut Contex
     let mut remapped_models = Vec::new();
     let mut remapped_fields = Vec::new();
     let mut reintrospected_id_names = Vec::new();
+    let mut models_without_identifiers = Vec::new();
+    let mut models_without_columns = Vec::new();
+    let mut unsupported_types = Vec::new();
```
Comment (tomhoule) [inline introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs:13]: Not a fan of all these local variables for different types of warnings. These should probably live in the context, and the code that extracts them at the end of the function be abstracted.

File: libs/dml/src/model.rs [hunk 2]
```diff
@@ -455,34 +455,19 @@ impl Model {
     /// optional unique fields are NOT considered a unique criteria
     /// used for: A Model must have at least one STRICT unique criteria.
     pub fn strict_unique_criterias(&self) -> Vec<UniqueCriteria> {
-        self.unique_criterias(false, false)
```
Comment (tomhoule) [inline libs/dml/src/model.rs:458]: Simplification made possible by the other changes.

File: introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs (hunk 1 above)
Comment (eviefp) [reply introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs:13]: Would a comment in the code help? Just to give a hint to whoever is here next and goes "ick, I should figure out how to make this better".

**Review by eviefp — APPROVED** [review]
lgtm :+1:

File: introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs (hunk 1 above)
Comment (tomhoule) [reply introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs:13]: Good point, will write on.

Comment (tomhoule) [reply introspection-engine/connectors/sql-introspection-connector/src/introspection/models.rs:16]: Done.

**Pull request merged**